src/Symfony/Component/Mailer/Transport/Smtp/Stream/AbstractStream.php

```diff
@@ -84,7 +84,7 @@ public function readLine(): string
             if ($metas['timed_out']) {
                 throw new TransportException(sprintf('Connection to "%s" timed out.', $this->getReadConnectionDescription()));
             }
-            if ($metas['eof']) {
+            if (\feof($this->out)) {
```

Comment:
```suggestion
            if (feof($this->out)) {
```

Maybe we should also add a comment so that this is not changed back again in the future.